Plugins/CGAL/vtkCGAL/vtkCGALRegionGrowing.cxx

```diff
@@ -0,0 +1,208 @@
+
+// -- VTK
+#include <vtkCellData.h>
+#include <vtkCGALUtilities.h>
+#include <vtkFieldData.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+#include <vtkIntArray.h>
+#include <vtkPointData.h>
+#include <vtkSmartPointer.h>
+#include <vtkTimerLog.h>
+
+// -- CGAL
+#include <CGAL/memory.h>
+#include <CGAL/IO/Color.h>
+#include <CGAL/Iterator_range.h>
+#include <CGAL/HalfedgeDS_vector.h>
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/Polyhedron_3.h>
+
+#include <CGAL/Shape_detection/Region_growing/Region_growing.h>
+#include <CGAL/Shape_detection/Region_growing/Region_growing_on_polygon_mesh.h>
+
+// -- STL
+#include <vector>
+#include <cstdlib>
+#include <iostream>
+#include <iterator>
+
+// Type declarations.
+using Kernel = CGAL::Exact_predicates_exact_constructions_kernel;
+
+using FT = typename Kernel::FT;
+using Point_3 = typename Kernel::Point_3;
+
+using Color = CGAL::Color;
+
+// Choose the type of a container for a polygon mesh.
+#define USE_SURFACE_MESH
+
+#if defined(USE_SURFACE_MESH)
+  using Polygon_mesh = CGAL::Surface_mesh<Point_3>;
+  using Face_range = typename Polygon_mesh::Face_range;
+
+  using Neighbor_query =
+    CGAL::Shape_detection::Polygon_mesh::One_ring_neighbor_query<Polygon_mesh>;
+  using Region_type =
+    CGAL::Shape_detection::Polygon_mesh::Least_squares_plane_fit_region<Kernel, Polygon_mesh>;
+  using Sorting =
+    CGAL::Shape_detection::Polygon_mesh::Least_squares_plane_fit_sorting<Kernel, Polygon_mesh, Neighbor_query>;
+#else
+  using Polygon_mesh =
+    CGAL::Polyhedron_3<Kernel, CGAL::Polyhedron_items_3, CGAL::HalfedgeDS_vector>;
+  using Face_range =
+    typename CGAL::Iterator_range<typename boost::graph_traits<Polygon_mesh>::face_iterator>;
+
+  using Neighbor_query =
+    CGAL::Shape_detection::Polygon_mesh::One_ring_neighbor_query<Polygon_mesh, Face_range>;
+  using Region_type =
+    CGAL::Shape_detection::Polygon_mesh::Least_squares_plane_fit_region<Kernel, Polygon_mesh, Face_range>;
+  using Sorting =
+    CGAL::Shape_detection::Polygon_mesh::Least_squares_plane_fit_sorting<Kernel, Polygon_mesh, Neighbor_query, Face_range>;
+#endif
+
+using Region = std::vector<std::size_t>;
+using Regions = std::vector<Region>;
+
+using Vertex_to_point_map = typename Region_type::Vertex_to_point_map;
+
+using Region_growing = CGAL::Shape_detection::Region_growing<Face_range, Neighbor_query, Region_type, typename Sorting::Seed_map>;
+
+// -----------------------------------------------------------------------------
+// Declare the plugin
+vtkStandardNewMacro(vtkCGALRegionGrowing);
+
+// -----------------------------------------------------------------------------
+// Constructor
+// Todo.
+vtkCGALRegionGrowing::vtkCGALRegionGrowing()
+{
+  this->MaxDistanceToPlane = 1;
+  this->MaxAcceptedAngle = 45;
+  this->MinRegionSize = 5;
+}
+
+// -----------------------------------------------------------------------------
+// Todo.
```

Comment:
Same, what's to do?

Reply:
same here, some description of the steps achieved in the RequestData is needed